nixpkgs/nixpkgs.bzl

```diff
@@ -175,14 +178,169 @@ def nixpkgs_package(*args, **kwargs):
     if "repositories" in kwargs:
         inversed_repositories = {value: key for (key, value) in kwargs["repositories"].items()}
         kwargs.pop("repositories")
-        _nixpkgs_package(
+        f(
             repositories = inversed_repositories,
             *args,
             **kwargs
         )
     else:
         _nixpkgs_package(*args, **kwargs)
 
+def _nixpkgs_packages_instantiate_impl(repository_ctx):
+    repository_ctx.file("BUILD", "exports_files([\"nix_attrs.nix\"])")
+
+    nix_instantiate_path = _executable_path(
+        repository_ctx,
+        "nix-instantiate",
+        extra_msg = "See: https://nixos.org/nix/",
+    )
+
+    packages_record_inside = ";".join(
+        [
+            "\"{name}\" = nixpkgs.{attrPath}".format(name = name, attrPath = attrPath)
+            for (name, attrPath) in repository_ctx.attr.packages.items()
+        ]
+    )
+    packages_record = "nixpkgs: { " + packages_record_inside + "; }"
+
+    repository_ctx.file("packages_attributes_mappings.nix", packages_record)
+
+    nix_set_builder = repository_ctx.template("drv_set_builder.nix", Label("@io_tweag_rules_nixpkgs//nixpkgs:drv_set_builder.nix"))
+    nix_instantiate_args = [
+        "--eval",
+        "--strict",
+        "--read-write-mode",
+        "--json",
+        "drv_set_builder.nix",
+        "--arg", "packages", "import ./packages_attributes_mappings.nix",
```

Comment:
in theory it should be possible to pass the package list straight as nix code in the argument and avoid the intermediary file. It also has the advantage that if the command it printed, all the context is available to run it manually.

Reply:
That's what I tried initially, but the original motivation for all that is to run it with the whole of `haskellPackages` which gave me `command-line too long` errors